File: Embarcadero/XE5/CPP/PhotoEditorDemo/MainFrm_Phone.cpp
//---------------------------------------------------------------------------

#include <fmx.h>
#pragma hdrstop

#include "MainFrm_Phone.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "MainFrm"
#pragma resource "*.fmx"
TPhoneMainForm *PhoneMainForm;
//---------------------------------------------------------------------------
__fastcall TPhoneMainForm::TPhoneMainForm(TComponent* Owner)
	: TBaseMainForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TPhoneMainForm::ActionListUpdate(TBasicAction *Action, bool &Handled)
{
	TBaseMainForm::ActionListUpdate(Action, Handled);
    TextToSelectImage->Visible = FRawBitmap->IsEmpty();
}
//---------------------------------------------------------------------------
void __fastcall TPhoneMainForm::ImageContainerClick(TObject *Sender)
{
	if (FRawBitmap->IsEmpty()) {
		ActionTakePhotoFromLibrary->ExecuteTarget(ButtonTakePhotoFromLibrary);
	}
}
//---------------------------------------------------------------------------

File: Embarcadero/XE5/ObjectPascal/PhotoEditorDemo/MainFrm_Phone.pas
unit MainFrm_Phone;

interface

uses
  System.SysUtils, System.Types, System.UITypes, System.Rtti, System.Classes,
  System.Variants, FMX.Types, FMX.Controls, FMX.Forms, FMX.Dialogs,
  FMX.StdCtrls, FMX.ActnList, FMX.MediaLibrary.Actions, System.Actions, FMX.StdActns, FMX.Objects,
  MainFrm, FMX.Effects, FMX.Layouts, FMX.Ani;

type
  TPhoneMainForm = class(TBaseMainForm)
    TextToSelectImage: TText;
    procedure ActionListUpdate(Action: TBasicAction; var Handled: Boolean);
    procedure ImageContainerClick(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  PhoneMainForm: TPhoneMainForm;

implementation

{$R *.fmx}

procedure TPhoneMainForm.ActionListUpdate(Action: TBasicAction; var Handled: Boolean);
begin
  inherited;
  TextToSelectImage.Visible := FRawBitmap.IsEmpty;
end;

procedure TPhoneMainForm.ImageContainerClick(Sender: TObject);
begin
  if FRawBitmap.IsEmpty then
    ActionTakePhotoFromLibrary.ExecuteTarget(ButtonTakePhotoFromLibrary);
end;

end.
